### PR TITLE
fix: Prevent Manually Added Content in CLI Docs to be Overwritten

### LINKED ISSUE #9104: [mesheryctl] Rework existing `mesheryctl` command ref docs script
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior

Currently, the docs for the `mesheryctl` commands are auto-generated using Cobra library's inbuilt functionality which enables to generate pages from variables in the program. However, the functionality is limited and can't perform below functions:
- Protection from overwrite (if new content is manually added from page instead of program) 
- Command reference table generation when the script is run
 Because of this, the action is not fully automated and not reliable due to data overwrite possibility 

### Desired Behavior
Allow contributors to offer additive documentation to that of the auto-generated documentation without it getting overwritten and without having to edit Golang in order to enhance the docs. Instead, they can stay within markdown and add additional context, usage, examples, screenshots, videos, and so forth.

### Implementation
A simple approach to achieving this is to create a START and END designation - a section of each CLI reference page as being untouched by the auto-generated script. See the mockup for a simple designation that would work very well.

### Acceptance Tests
1. No loss of hand-written docs. The logic inside the script needs to be updated to touch anything between this section and the start of the next section. 
1. Meshery Docs needs to be updated to include instructions in the Contributing page for Meshery CLI. 
1. Each CLI reference page should have an Info alert that informs contributors of this framework and points them to the Meshery CLI contributing doc for an explanation.

### Mockup
```
<!-- ###########################################################################

                 START OF MANUAL DOCUMENTATION
IT IS SAFE TO ADD ADDITIONAL USAGE AND EXAMPLES BELOW THIS LINE.
MORE INFO: https://docs.meshery.io/project/contributing/contributing-cli

############################################################################ -->

### Sample Usage 

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. 

\`\`\`
mesheryctl version --config docker-compose.yaml
\`\`\`

Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

![screenshot of meshery](https://raw.githubusercontent.com/meshery/meshery/master/.github/assets/images/meshery/meshery-logo-dark-text-side.png)

<!-- ###########################################################################

                 END OF MANUAL DOCUMENTATION

############################################################################ -->

```

---

### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](http://discuss.meshery.io) and [Layer5 Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by aboobakersiddiqr63**
**Notes for Reviewers**

This PR fixes #9104 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ✓] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 73d569d231cb0408b90d13be2e8acaddacea3e29 at: https://652e0e5a9bb75234d233d311--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 3977467d338735dce77423bfeb1c69ead6ed277c at: https://652e1350165c1d3b995e5577--meshery-docs-preview.netlify.app

**Comment by saurabh100ni** [comment]
Thanks for raising the PR, let's Discuss this in the Meshery Development Call today,
Please add this as an agenda in the Doc : [Doc](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit)

**Comment by l5io** [comment]
🚀 Preview for commit d88c6dc8228a367830a41b148f7b5725e61c2abe at: https://652fbe74e46c95112af055f2--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit b43c313cb332c381ead1ea8d7a4215bc5cb41ce9 at: https://652fc5e21989fd1b7600dc93--meshery-docs-preview.netlify.app

**Review by alphaX86 — COMMENTED** [review]
Great effort. Curious though. So to invoke this user would be applying them on the markdown page or in go program?

**Comment by aboobakersiddiqr63** [comment]
> Great effort. Curious though. So to invoke this user would be applying them on the markdown page or in go program?

Thanks @alphaX86. The user just needs to add the comment on the md file

**Comment by l5io** [comment]
🚀 Preview for commit 997e5ebac94f1622ea07384895ee2411550a1e90 at: https://6533be2256d42a1ab87ea689--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 1322fda62d4f6f9b48cca583b92cbae173937579 at: https://6533d503e7d324284f546288--meshery-docs-preview.netlify.app

File: docs/pages/project/contributing/contributing-cli.md
Comment (leecalcote) [inline docs/pages/project/contributing/contributing-cli.md:207]:
```suggestion
## Preserving Manually Added Documentation

`mesheryctl` uses Cobra CLI and GitHub Actions to automate the generation of command documentation. On occasion, additional documentation beyond that included in the `mesheryctl` Golang files is ideal to capture and include in the CLI reference pages. Contributors are encouraged to add more usage examples, screenshots, video explainers and so forth to any of the CLI reference pages. To protect any manually added content and ensure it remains intact after regeneration, create a separate Jekyll `include` file. Follow file naming scheme outlined below:

```
{% raw %}{% include mesheryctl/<name-of-command> %}{% endraw %}
```

This format should reference an external file where your manual changes are stored. Any content added using this method will not be altered during the documentation generation process, but instead will be included post-auto doc generation. When making new changes or additions, understand that these additional details are positioned at the end their given CLI reference page, so bear this in mind as you organize and present your additional command details.

```

File: mesheryctl/doc/doc.go
Comment (leecalcote) [inline mesheryctl/doc/doc.go:218]:
```suggestion
		buf.WriteString("Go back to [command reference index](/reference/mesheryctl/), if you want to add content manually to the CLI documentation, please refer to the [instruction](/project/contributing/contributing-cli#preserving-manually-added-documentation) for guidance.")
```

**Review by leecalcote — APPROVED** [review]
Looking good. 👍 

I think this new `include` method is a bit cleaner. Nice work.

**Comment by saurabh100ni** [comment]
Let's discuss this on the Meshery dev call tomorrow at 7:30 pm IST / 9:00 am CT,
Please add this as your agenda item in Docs: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit

**Comment by leecalcote** [comment]
@aboobakersiddiqr63 will you send word when you have added an example to the Contributing portion of this PR?

**Comment by aboobakersiddiqr63** [comment]
> @aboobakersiddiqr63 will you send word when you have added an include file example to the Contributing portion of this PR?

@leecalcote added an example of how to add manual content to the cli docs and updated other changes you have mentioned

**Comment by l5io** [comment]
🚀 Preview for commit f34721d2363b2cfcd5ada6171b9c20d72b8ca630 at: https://653a162b65721959d62c954e--meshery-docs-preview.netlify.app

**Comment by alphaX86** [comment]
> > @aboobakersiddiqr63 will you send word when you have added an include file example to the Contributing portion of this PR?
> 
> @leecalcote added an example of how to add manual content to the cli docs and updated other changes you have mentioned

I could not find an example actually. Could you share any markdown file utilizing this ability?

**Comment by aboobakersiddiqr63** [comment]
@alphaX86 

> > > @aboobakersiddiqr63 will you send word when you have added an include file example to the Contributing portion of this PR?
> > 
> > 
> > @leecalcote added an example of how to add manual content to the cli docs and updated other changes you have mentioned
> 
> I could not find an example actually. Could you share any markdown file utilizing this ability?

@alphaX86 I have added the example to contributing-cli.md file on how to proceed 
image

**Comment by alphaX86** [comment]
@aboobakersiddiqr63 noted. I'll try this and get back to you then 👍

**Comment by leecalcote** [comment]
Gents, do we have an example for the docs?

Good to merge after this, yes?

**Comment by abhijeetgauravm** [comment]
Hey @aboobakersiddiqr63  Let's discuss this on Meshery Development Meeting tomorrow at 8:30 PM IST / 9 AM Central time. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#).

**Comment by MUzairS15** [comment]
@alphaX86 Can you finish this off. // @aboobakersiddiqr63

**Comment by alphaX86** [comment]
> @alphaX86 Can you finish this off. // @aboobakersiddiqr63

I'll wrap this up @MUzairS15

**Comment by l5io** [comment]
🚀 Preview for commit 979747d6e0d56dfb85983e5fe5585e34811fa09a at: https://655a10c9040c8637e4340e96--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
> > @alphaX86 Can you finish this off. // @aboobakersiddiqr63
> 
> I'll wrap this up @MUzairS15

@alphaX86 good to go?

**Comment by alphaX86** [comment]
> > > @alphaX86 Can you finish this off. // @aboobakersiddiqr63
> > 
> > I'll wrap this up @MUzairS15
> 
> @alphaX86 good to go?

I'm testing the changes @leecalcote. Once they are good, we're good to merge this by EOD.

**Comment by l5io** [comment]
🚀 Preview for commit bfb86114e8395cabfe67ea840db616844634c0d7 at: https://655e27ef2a036247e6b1d648--meshery-docs-preview.netlify.app

**Review by alphaX86 — DISMISSED** [review]
Tested the changes 🟢. Made changes in doc page as well. LGTM 👍

**Comment by alphaX86** [comment]
Hold... one bug found in docs. Fixing it quick...

**Comment by l5io** [comment]
🚀 Preview for commit 32a9aa9f4fcdd73e7122347e779aacc274a30096 at: https://655e333de2e9a44fe798016e--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 13a9e0a9c36f64470ed678e80fddedf09dd64c4d at: https://655e34cf2a03625285b1d70e--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit b623613241e962d56dfa6524510ae9f138d3f259 at: https://655e37c4b2a3c44dee74d4fe--meshery-docs-preview.netlify.app

**Review by alphaX86 — APPROVED** [review]

**Pull request merged**
